subsys/bluetooth/host/l2cap.c

```diff
@@ -1971,12 +1971,11 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 		BT_WARN("Unable to send seg %d", err);
 		atomic_inc(&ch->tx.credits);
 
-		/* If the segment is not the original buffer release it since it
-		 * won't be needed anymore.
+		/* Release the segment, even if it is the original buf. the
+		 * caller of this function already holds a reference to this
+		 * buffer and l2cap_chan_le_send will get called again.
```

Comment:
Do I understand correctly?: `l2cap_chan_create_seg` returns an owned reference, if any. `bt_l2cap_send_cb`, if successful, moves the reference, but otherwise not. So, when it was not successful, we must still move the reference (into `net_buf_unref`) to avoid leaking it.